### DISCUSSION TITLE
Adding a easy way of implementing `get_configuration_warnings` to make hierarchy dependency easier.

### DISCUSSION THREAD

**Discussion by fenilli** [godotengine/godot-proposals#6306] [GUI]
I think a cool system that godot has, is nodes that have a heavy dependency will give warnings in the GUI letting you know it needs a child of a specific node type, right now it is not possible by my knowledge to create a Scene with this behaviour without having to use `tool` into a script, this is really a bummer as it will lead to more code and possible problems in the editor.

So I would think a cool implementation would be to add a way to implement an annotation like `@export` to have `@warning` to be able to give those warnings in the GUI.

**Comment by Mickeon** [comment]
A selective `@tool`-like annotation that allows selectively enabling methods to be executed in the editor would also do the trick and be more useful overall.

**Reply by fenilli** [reply]
Agree, if it could be selective, the problem with tool itself, it is globally to that script, so it would also need more work I believe.
